pkg/plugin/plugin_agent.go

```diff
@@ -0,0 +1,126 @@
+// SPDX-FileCopyrightText: Cristian Mogildea
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package plugin
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/pkg/stdcopy"
+)
+
+func StartContainer(cfg *Config) error {
```

Comment:
The  function says that it starts a container, but in reality it does more, e.g. starting a HTTP server. Single responsibility.